WalletCore/Classes/ViewModel/TransactionFilterViewModel.swift

```diff
@@ -59,6 +63,59 @@ open class TransactionFilterViewModel {
             .map({ $0.end })
             .bind(to: endDate)
             .disposed(by: disposeBag)
+        
+        self.errors = errorsSubject.asDriver(onErrorJustReturn: [:])
+    }
+}
+
+public extension TransactionFilterViewModel {
+    var startDateValue: Date? {
+        
+        get {
+            return startDate.value
+        }
+        
+        set {
+            var check = self.compareDate(startDate: newValue, withDate:endDateValue)
+            
+            if(check) {
+                startDate.value = newValue
+            }
+            else {
+                self.errorsSubject.onNext(["Message": Localize("filter.newDesign.error.message.startDate")])
+            }
+        }
+    }
+    
+    var endDateValue: Date? {
+        
+        get {
+            return endDate.value
+        }
+        
+        set {
+            var check = self.compareDate(startDate: startDateValue, withDate: newValue)
+            
+            if(check) {
+                endDate.value = newValue
+            }
+            else {
+                self.errorsSubject.onNext(["Message": Localize("filter.newDesign.error.message.endDate")])
+            }
+            
+        }
+    }
+}
+
+//Date validation
+public extension TransactionFilterViewModel {
+    
+    func compareDate(startDate: Date?, withDate endDate:Date?) -> Bool {
```

Comment:
Please move this logic in Date extension, so can be used across the whole module